packages/config-resolver/src/components.ts

```diff
@@ -155,7 +165,7 @@ export namespace EndpointsConfig {
     /**
      * The endpoint provider to call if no endpoint is provided
      */
-    endpointProvider?: any;
+    endpointProvider?: (region: string, options?: any) => HttpEndpoint;
```

Comment:
Leave public interfaces' options as `any` on purpose as we want it to be private API